### DISCUSSION TITLE
Enrolling new chains to Optics

### DISCUSSION THREAD

**Discussion by anna-carroll** [celo-org/optics-monorepo#609] [Ideas]
To add another chain to Optics, 

**Setup on the new chain**
- deploy a Home contract on the new chain 
- deploy a Replica for **each** existing remote Home on the new chain 
- deploy a UsingOptics contract on the new chain
   - enroll all of the newly deployed Replicas for existing remote Homes
- deploy a GovernanceRouter contract on the new chain
   - enroll all of the existing GovernanceRouters on remote chains 
- transfer ownership of Home, Replicas, and Upgrade Beacons to GovernanceRouter
- transfer ownership of UsingOptics (to whomst?)

**Setup on each other chain**
- deploy a Replica for the new Home
- transfer ownership of the Replica to the GovernanceRouter
- enroll the new local Replica with the canonical UsingOptics 
- enroll the new remote GovernanceRouter with the local GovernanceRouter
-----

Documentation, Deploy, and/or Tutorial

**Comment by prestwich** [comment]
should the global governance be responsible for _every_ replica? 🤔

**Comment by anna-carroll** [comment]
> should the global governance be responsible for _every_ replica? 🤔

I should think so?


> We probably need to account for ability to add a new chain within the Governance Xapp

This is easier, we can just use some combination fo the `Call` message(s) to deploy a contract and do the necessary setup

> We probably want to create a Replica Contract Factory on each chain

Note: 👁️ Dharma Smart Wallet Factory

**Comment by prestwich** [comment]
> > should the global governance be responsible for _every_ replica? 🤔
> 
> I should think so?

let me think about this one more 🤔
